olivertwist/main.py

```diff
@@ -79,21 +80,30 @@ def check(input, config, html=True, browser=False):
 
 def format_for_terminal(results: List[Result]):
     for result in results:
-        colour = "red" if result.has_failures else "green"
+        colour = get_colour(result)
         name = click.style(f"{result.rule.name}:", fg=colour)
         link = click.style(
             f"http://olivertwi.st/rules/#{result.rule.id}",
             fg="blue",
         )
         click.echo(f"{name} [{link}]:")
         for node in result.failures:
-            click.secho(f" - {node.id}", fg="red")
+            click.secho(f" - {node.id}", fg=colour)
 
         click.echo()
 
 
+def get_colour(result: Result):
+    colour = "green"
+    if result.has_errors:
+        colour = "red"
+    if result.has_warnings:
```

Comment:
I didn't spot this before, but this order means that any warning will downgrade the colour from `red` to `yellow`. Probably not an issue while rule severity is set universally but could bite us in future. How about:

```python
def get_colour(result: Result):
    if result.has_errors:
        return "red"
    elif result.has_warnings:
        return "yellow"
    return "green"
```